lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -352,6 +356,102 @@ func (c *ContainerProxy) ContainerRunning(vc *viccontainer.VicContainer) (bool,
 	return inspectJSON.State.Running, nil
 }
 
+func (c *ContainerProxy) Wait(vc *viccontainer.VicContainer, timeout time.Duration) (exitCode int32, processStatus string, containerState string, reterr error) {
+	defer trace.End(trace.Begin(vc.ContainerID))
+
+	if vc == nil {
+		reterr = InternalServerError("Wait bad arguments")
+		return
+	}
+
+	// Get an API client to the portlayer
+	client := PortLayerClient()
+	if client == nil {
+		reterr = InternalServerError("Wait failed to create a portlayer client")
+		return
+	}
+
+	params := containers.NewContainerWaitParamsWithContext(ctx).
+		WithTimeout(int64(timeout.Seconds())).
+		WithID(vc.ContainerID)
+	results, err := client.Containers.ContainerWait(params)
+	if err != nil {
+		switch err := err.(type) {
+		case *containers.ContainerWaitNotFound:
+			// since the container wasn't found on the PL lets remove from the local
+			// cache
+			cache.ContainerCache().DeleteContainer(vc.ContainerID)
+			reterr = NotFoundError(vc.ContainerID)
+			return
+		case *containers.ContainerWaitInternalServerError:
+			reterr = InternalServerError(err.Payload.Message)
+			return
+		default:
+			reterr = InternalServerError(err.Error())
+			return
+		}
+	}
+
+	if results == nil || results.Payload == nil {
+		reterr = InternalServerError("Unexpected swagger error")
+	}
+
+	containerInfo := results.Payload
+
+	return *containerInfo.ProcessConfig.ExitCode, *containerInfo.ProcessConfig.Status, *containerInfo.ContainerConfig.State, nil
+}
+
+func (c *ContainerProxy) Signal(vc *viccontainer.VicContainer, sig uint64) error {
+	defer trace.End(trace.Begin(vc.ContainerID))
+
+	if vc == nil {
+		return InternalServerError("Signal bad arguments")
+	}
+
+	// Get an API client to the portlayer
+	client := PortLayerClient()
+	if client == nil {
+		return InternalServerError("Signal failed to create a portlayer client")
+	}
+
+	if running, err := c.ContainerRunning(vc); !running {
+		if err == nil {
```

Comment:
idomatic would be  collapsing this to a single test clause: `!running && err == nil`